File: web/backend/models/Location.js
const mongoose = require('mongoose');

const locationSchema = mongoose.Schema(
  {
    clothingStoreId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'ClothingStore',
    },
      address: {
        type: String,
      },
      city: {
        type: String,
      },
      country: {
        type: String,
        default: 'Slovenia',
      },
      coordinates: {
        // GeoJSON format for MongoDB geospatial queries
        type: {
          type: String,
          enum: ['Point'],
          default: 'Point',
        },
        coordinates: {
          // [longitude, latitude]
          type: [Number],
          required: true,
        }
      },    
  }
);

// geospatial index for efficient location-based queries
locationSchema.index({ 'location.coordinates': '2dsphere' });

const Location = mongoose.model('Location', locationSchema);
module.exports = Location;;

File: DesktopApp/src/main/kotlin/models/Location.kt
package models

import kotlinx.serialization.Serializable
import kotlinx.serialization.SerialName

@Serializable
data class Location(
    @SerialName("_id")
    val id: String? = null,
    val clothingStoreId: ClothingStoreRef,
    val address: String,
    val city: String,
    val country: String,
    val coordinates: Coordinates? = null
)

@Serializable
data class ClothingStoreRef(
    @SerialName("_id")
    val id: String,
    val name: String,
    val website: String? = null
)

@Serializable
data class Coordinates(
    val type: String = "Point",
    val coordinates: List<Double> // [longitude, latitude]
)
